### PR TITLE
Build and publish terraform provider snowflake

### PR DESCRIPTION
**Pull request by wookasz**
## What
- Switch to managing dependencies with go modules (this is in-line with upstream)
- Add Dockerfile
- Build image with provider binary in codefresh

## Why
This allows projects to share the provider rather than having to build from source each time

## Testing
NMTN - we're just building in a docker container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COPY go.mod .
+COPY go.sum .
+
+RUN gofmt -l `find . -name '*.go'`
+RUN GO111MODULE=on go build
+
+# back to the base-ics, we only need the binary
+FROM base
+
+ENV BIN_DIR=/go/src/github.com/sagansystems/terraform-provider-snowflake
```
Comment (mkj28) [inline Dockerfile]: do you need this env variable? just curious... if not - you can hardcode entire path in COPY

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+WORKDIR /usr/local/bin
```
Comment (mkj28) [inline Dockerfile]: LOL :D

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+COPY --from=builder ${BIN_DIR}/terraform-provider-snowflake .
```
Comment (mkj28) [inline Dockerfile]:
```suggestion
COPY --from=builder /go/src/github.com/sagansystems/terraform-provider-snowflake/terraform-provider-snowflake /usr/local/bin/
```

File: Dockerfile (hunk 1 above)
Comment (wookasz) [reply Dockerfile]: no, the line just got a bit lengthy to read, but I can make it one line

File: Dockerfile (hunk 3 above)
Comment (mkj28) [reply Dockerfile]: and remove the ENV and WORKDIR

Comment (mkj28) [reply Dockerfile]: NIT tho

**Review by mkj28 — APPROVED** [review]

**Pull request merged**